frame/proxy/src/lib.rs

```diff
@@ -171,19 +167,17 @@ decl_module! {
 				.find(|x| &x.0 == &who && force_proxy_type.as_ref().map_or(true, |y| &x.1 == y))
 				.ok_or(Error::<T>::NotProxy)?;
 
-			// We're now executing as a freshly authenticated new account, so the previous call
-			// restrictions no longer apply.
-			let _clear_guard = ClearFilterGuard::<T::IsCallable, <T as Trait>::Call>::new();
-			let _filter_guard = FilterStackGuard::<T::IsCallable, <T as Trait>::Call>::new(
-				move |c| match c.is_sub_type() {
+			// This is a freshly authenticated new account, the origin restrictions doesn't apply.
+			let mut origin: T::Origin = frame_system::RawOrigin::Signed(real).into();
+			origin.add_filter(move |c: &<T as frame_system::Trait>::Call| {
```

Comment:
we need a way of not just adding a filter but of temporarily cancelling all previously added filters.

Reply:
ah indeed, this can be done with `origin.reset_filter(); origin.add_filter...` but will do with new API

Reply:
hmm actually it is already doing so, this origin here is created with BaseCallFilter and the proxy filter is added on top of it, it doesn't use the filter contained in the origin of this dispatch.

the filter for the proxy call should be BaseCallFilter && ProxyAddedFilter isn't it ?

Reply:
yeah - in any case, if the tests pass, we're good.